src/tactic/rewrite_all/congr.lean

```diff
@@ -0,0 +1,133 @@
+-- Copyright (c) 2018 Scott Morrison. All rights reserved.
+-- Released under Apache 2.0 license as described in the file LICENSE.
+-- Authors: Scott Morrison, Keeley Hoek
+import tactic.rewrite_all.common
+
+namespace tactic
+
+-- Sometimes `mk_congr_arg` fails, when the function is 'superficially dependent'.
+-- This hack `dsimp`s the function before building the `congr_arg` expression.
+-- Unfortunately it creates some dummy hypotheses that I can't work out how to dispose of cleanly.
+meta def mk_congr_arg_using_dsimp (G W : expr) (u : list name) : tactic expr :=
+do
+  s ← simp_lemmas.mk_default,
```

Comment:
put `s ← simp_lemmas.mk_default,` on the same line as `do`

Reply:
Fixed.